variables.tf

```diff
@@ -92,3 +92,27 @@ variable "drg_display_name" {
   type        = string
   default     = null
 }
+
+# routing rules
+
+variable "internet_gateway_route_rules" {
+  description = "(Updatable) List of routing rules to add to Internet Gateway Route Table"
+  type = list(object({
+    destination       = string
+    destination_type  = string
+    network_entity_id = string
+    description       = string
+  }))
+  default = null
+}
+
+variable "nat_gateway_route_rules" {
```

Comment:
See above for internet_gateway_rules

Reply:
destination, destination_type and network_entity_id are required by API. Only description is optional.
So I think having a descriptive schema is less guesswork for the user and enforce what the api expect.